components/console-backend-service/internal/domain/servicecatalogaddons/clusteraddonsconfiguration_service.go

```diff
@@ -0,0 +1,404 @@
+	idxURLs := map[string]struct{}{}
+	for _, u := range urls {
+		idxURLs[u] = struct{}{}
+	}
+
+	var result []v1alpha1.SpecRepository
+	for _, r := range repos {
+		if _, found := idxURLs[r.URL]; !found {
+			result = append(result, r)
+		}
+	}
+	return result
+}
+
+func (s *clusterAddonsConfigurationService) toMapLabels(givenLabels *gqlschema.Labels) map[string]string {
+	if givenLabels == nil {
+		return nil
+	}
+
+	labels := map[string]string{}
+	for k, v := range *givenLabels {
+		labels[k] = v
+	}
+	return labels
+}
+
+func (s *clusterAddonsConfigurationService) toSpecRepositories(urls []string) []v1alpha1.SpecRepository {
+	var repos []v1alpha1.SpecRepository
+	for _, u := range urls {
+		repos = append(repos, v1alpha1.SpecRepository{URL: u})
+	}
+
+	return repos
+}
+
+func (s *clusterAddonsConfigurationService) getClusterAddonsConfiguration(name string) (*v1alpha1.ClusterAddonsConfiguration, error) {
+	item, exists, err := s.addonsCfgInformer.GetStore().GetByKey(name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while getting %s %s", pretty.AddonsConfiguration, name)
+	}
+
+	if !exists {
+		return nil, errors.Errorf("%s doesn't exists", name)
+	}
+
+	addons, ok := item.(*v1alpha1.ClusterAddonsConfiguration)
+	if !ok {
+		return nil, fmt.Errorf("incorrect item type: %T, should be: *v1alpha1.ClusterAddonsConfiguration", item)
+	}
+
+	return addons, nil
+}
+
+func (svc *clusterAddonsConfigurationService) Subscribe(listener resource.Listener) {
+	svc.addonsNotifier.AddListener(listener)
+}
+
+func (svc *clusterAddonsConfigurationService) Unsubscribe(listener resource.Listener) {
+	svc.addonsNotifier.DeleteListener(listener)
+}
+
+// Deprecated, the ClusterAddonsConfiguration should be used instead
+func (s *clusterAddonsConfigurationService) ListConfigMaps(pagingParams pager.PagingParams) ([]*v1.ConfigMap, error) {
+	items, err := pager.From(s.cfgMapinformer.GetStore()).Limit(pagingParams)
+	if err != nil {
+		return nil, err
+	}
+
+	var configMaps []*v1.ConfigMap
+	for _, item := range items {
+		configMap, ok := item.(*v1.ConfigMap)
+		if !ok {
+			return nil, fmt.Errorf("incorrect item type: %T, should be: *ConfigMap", item)
+		}
+
+		configMaps = append(configMaps, configMap)
+	}
+
+	return configMaps, nil
+}
+
+// Deprecated, the ClusterAddonsConfiguration should be used instead
+func (s *clusterAddonsConfigurationService) AddReposToConfigMap(name string, urls []string) (*v1.ConfigMap, error) {
+	configMap, err := s.getConfigMapAddonsConfiguration(name)
+	if err != nil {
+		return nil, err
+	}
+
+	configMap.Data[addonsCfgKey] = s.joinURLs(configMap.Data[addonsCfgKey], urls)
+
+	result, err := s.cfgMapClient.Update(configMap)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while updating %s %s", pretty.AddonsConfiguration, configMap.Name)
+	}
+
+	return result, nil
+}
+
+func (s *clusterAddonsConfigurationService) joinURLs(base string, newUrls []string) string {
```

Comment:
You may add a `deprecated` comment above these functions: `jointURLs(), removeURL(), setLabels()`

Reply:
I've added the `deprecated` comment only for the external methods. We do not require that for helper functions. 

Here it will be removed: https://github.com/mszostok/kyma/pull/888/files?w=1